warehouse/query-core/src/main/java/datawave/query/postprocessing/tf/TFFactory.java

```diff
@@ -61,9 +61,15 @@ public static com.google.common.base.Function<Tuple2<Key,Document>,Tuple3<Key,Do
         if (termFrequencyFieldValues.isEmpty()) {
             return new EmptyTermFrequencyFunction();
         } else {
-            TermOffsetPopulator offsetPopulator = new TermOffsetPopulator(termFrequencyFieldValues, config.getContentExpansionFields(),
-                            config.getEvaluationFilter(), config.getSource());
-            return new TermOffsetFunction(offsetPopulator, config.getTfFields());
+            
+            DocumentKeysFunction docKeyFunction = null;
+            
+            if (config.isTld()) {
+                docKeyFunction = new DocumentKeysFunction(config);
+            }
```

Comment:
why tld only?

Reply:
An event query by definition is for a single document, so the set of document keys will only be size one. 

TLD on the other hand could have many child document keys, and intersecting those before aggregating tfs will reduce the amount of work required.